spec/requests/api/v0/highlights_controller_spec.rb

```diff
@@ -5,6 +5,63 @@
 
   before { allow(Rails.application.config).to receive(:consider_all_requests_local) { false } }
 
+  describe 'GET /highlights' do
+    let!(:highlight1) { create(:highlight, user_uuid: user_uuid) }
+    let!(:highlight2) { create(:highlight, user_uuid: user_uuid) }
+    let!(:highlight3) { create(:highlight) }
+
+    let(:per_page) { 10 }
+    let(:query_params) do
+      {
+        source_type: 'openstax_page',
+        source_parent_ids: highlight1.source_parent_ids,
+        color: '#000000',
+        page: 1,
+        per_page: per_page,
+        order: 'desc'
+      }
+    end
+
+    context 'when the user is not logged in' do
+      context 'when the highlight does exist' do
+        it('does not allow getting highlights') do
+          get highlights_path, params: query_params
+          expect(response).to have_http_status(:unauthorized)
+        end
+      end
+    end
+
+    context 'when the user is logged in' do
+      before do
+        stub_current_user_uuid(user_uuid)
+      end
+
+      it 'gets the highlights for the user' do
+        get highlights_path, params: query_params
+        expect(response).to have_http_status(:ok)
+
+        highlights_json = json_response[:highlights]
```

Comment:
is this still json? I think it is just an array of hashes, so `highlights_array` or `highlights` might be better

Reply:
it was json before json_response converted it :)  will rename to highlights (more rubyish)